io/plugins/eu.esdihumboldt.hale.io.gml/src/eu/esdihumboldt/hale/io/gml/reader/internal/wfs/WfsBackedGmlInstanceCollection.java

```diff
@@ -412,6 +454,24 @@ private void createNextIterator() {
 				throw new IllegalStateException(e.getMessage(), e);
 			}
 
+			// Download all the files into a folder for Bucket service
+			if (tmpdir != null) {
+				try {
+					URL url = nextUri.toURL();
+					StringBuilder fileName = new StringBuilder(tmpdir);
+					fileName.append("/").append(url.getPath().replaceAll("/", "."))
```

Comment:
In principle OK for me, important is that it works fine with whatever strategy you use for handling it on the bucket-service side.

Did you check how the name was derived before when downloading via WFS in the bucket-service?

Reply:
The file was always provided from [bucket-service](https://github.com/wetransform/bucket-service/blob/baf3eddb03133661f078ccf9071d40dcc4b00e90/play/app/wfs/WfsImporter.scala#LL144C16-L144C16) to hale studio as `FileIOSupplier`. Which is [derived here](https://github.com/wetransform/bucket-service/blob/baf3eddb03133661f078ccf9071d40dcc4b00e90/play/app/controllers/Buckets.scala#LL1102C12-L1102C12) from queryMap and then reduced in the length.